src/Node/Expression/Variable/TemplateVariable.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig\Node\Expression\Variable;
+
+use Twig\Compiler;
+use Twig\Node\Expression\NameExpression;
+
+final class TemplateVariable extends NameExpression
+{
+    public function compile(Compiler $compiler): void
+    {
+        if ('_self' === $this->getAttribute('name')) {
+            $compiler->raw('$this');
+        } else {
+            $compiler
+                ->raw('$macros[')
```

Comment:
Maybe we can rename this variable as it stores templates, not macros.

Reply:
If those templates are only the ones used for macros, maybe `macroTemplates` ?

Reply:
I've just had a closer look and it can break extensions, like cache, so let's keep it like it is today for now.